_drafts/_draft-2024-02-22-connectivity-metrics.md

```diff
@@ -0,0 +1,232 @@
+
+Protocol analyzers offer a more sophisticated approach to connectivity monitoring, enabling developers to delve deeper into the intricacies of communication protocols. These tools provide detailed insights into packet structure, message sequences, and protocol validation. Protocol analyzers consist of hardware to capture and process the signals and software to visualize and further analyze the captured packets. Dedicated hardware and software can be a worthwhile investment, but these are pricey solutions. Check their capabilities and software compatibility against what you need; you would be surprised how often only older versions of Windows are supported. For 90% of us, some excellent free software and affordable hardware will suffice. With regards to software, I absolutely must recommend Wireshark. Wireshark is an incredibly powerful tool that captures and analyzes WiFi + networking protocols (IP, TCP, UDP, MQTT, etc), USB, and even BLE. This tool has saved me tons of effort and helped me numerous times. I’ve used Wireshark to find a bug in a BLE stack, profile MQTT connections, and improve my USB throughput, to name a few instances. It’s insane that it’s free, and I’d pay (practically) any price they offered. Here’s an example of a capture I collected to find a problem in my BLE stack:
+
+<p align="center">
+  <img width="650" src="{% img_url connectivity-metrics/wireshark-ble-capture.png %}" alt="BLE Packet Capture in Wireshark" />
+</p>
+
+For this problem, I observed lower throughput than expected during data transfer. Logs showed that the connection parameters were set correctly initially, but the data rate did not match this. Inspecting the capture in Wireshark led me to discover that two different connection parameter update methods in the BLE stack clobbered values set by the other, resulting in decreased throughput.
+
+Depending on the protocol, you may not need additional hardware. Wireshark can use your WiFi interface’s monitor or promiscuous mode to capture any signals in the air. It can also hook into your local Ethernet and USB interfaces for packet capture. For BLE sniffing hardware, I reach for my closest nRF52 dev kit! Nordic Semiconductor offers a BLE Sniffer package to download that includes plugins for Wireshark. With these, Wireshark will use your dev kit as an interface to capture packets from.
+
+**Strengths**:
+
+- Incredibly detailed analysis of packets on the wire/over the air
+- Great for investigating specific protocol issues
+
+**Best Used For**:
+
+- Single devices
+- Local debugging/investigation
+- Protocol errors and profiling specific devices
+
+Use protocol analyzers when you need extreme detail for a **single** device at your bench or desk. These tools will not help in the field unless you’re present. They should rarely be used because packet captures are very large and process-intense. Packet captures focus on a single device and can’t be used to aggregate across a growing fleet.
+
+### Attributes
```

Comment:
I think you could move attributes after metrics as a final tack-on technique since we have a bit less content for it. Also your transition sentences kind of set it up well (you mention metrics already in this first sentence)